packages/react-router5/modules/RouteProvider.js

```diff
@@ -16,24 +16,33 @@ class RouteProvider extends React.PureComponent {
         super(props)
         const { router } = props
 
+        state = {
+            mounted: false
+        }
         this.router = router
-        this.state = {
+        this.routeState = {
```

Comment:
we do not copy the props to the state anymore, this is considered a bad practice https://medium.com/@justintulk/react-anti-patterns-props-in-initial-state-28687846cc2e